vm/src/stdlib/math.rs

```diff
@@ -657,6 +657,34 @@ fn math_remainder(x: IntoPyFloat, y: IntoPyFloat, vm: &VirtualMachine) -> PyResu
     Ok(x)
 }
 
+#[derive(FromArgs)]
+struct ProdArgs {
+    #[pyarg(positional)]
+    iterable: PyIterable<PyObjectRef>,
+    #[pyarg(named, optional)]
+    start: OptionalArg<PyObjectRef>,
+}
+fn math_prod(args: ProdArgs, vm: &VirtualMachine) -> PyResult<PyObjectRef> {
+    let iter = args.iterable;
+
+    let mut result = match args.start {
+        OptionalArg::Present(start) => start,
+        OptionalArg::Missing => vm.new_pyobj(1),
+    };
+
+    for obj in iter.iter(vm)? {
+        let obj = obj?;
+        if let Ok(Some(_v)) = float::try_float_opt(&obj, vm) {
```

Comment:
`try_float_opt` is only used for floats. this functino shouldn't be called for general path.